marlowe-playground-client/src/Simulation/State.purs

```diff
@@ -176,28 +182,37 @@ updatePossibleActions oldState =
 
     (Tuple nextState actions) = extractRequiredActionsWithTxs txInput state contract
 
-    actionInputs = foldl (\acc act -> insertTuple (actionToActionInput nextState act) acc) mempty actions
+    usefulActions = mapMaybe removeUseless actions
+
+    actionInputs = foldl (\acc act -> insertTuple (actionToActionInput nextState act) acc) mempty usefulActions
```

Comment:
Anytime you see `foldl ... mempty` you probably want `fold` or `foldMap`.

Reply:
actually in this case its simpler, `Map.fromFoldable $ map`